src/CiviCrmApi.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Drupal\civicrm_entity;
+
+use Drupal\civicrm\Civicrm;
+
+class CiviCrmApi implements CiviCrmApiInterface {
+
+  /**
+   * Constructs a new CiviCrmApi object.
+   *
+   * @param \Drupal\civicrm\Civicrm $civicrm
+   *   The CiviCRM service.
+   */
+  public function __construct(Civicrm $civicrm) {
+    // Ensure CiviCRM is loaded and our function is available.
+    if (!function_exists('civicrm_api3')) {
+      $civicrm->initialize();
```

Comment:
Don't do this in the service constructor! It'd be better to call `$civicrm->initialize()` when we're about to do work with CiviCRM. Otherwise, we're initializing CiviCRM any time this service is constructed, and it might be constructed just to resolve a dependency, but when we're not actually going to do anything with CiviCRM.

For example, querying info about the entity type could lead to the entity type plugin being loaded and then initializing CiviCRM, but maybe it was just some generic Drupal code trying to some generic entity type info and not really load anything? Initializing CiviCRM is on par with bootstrapping Drupal and shouldn't be done unless we really need to!

I'd suggest putting this in a helper method and then call that at the top of each of the methods that do stuff like `get()`, `delete()`, etc, so it's only initialized just before actually doing something.